packages/base/src/commands.ts

```diff
@@ -1181,6 +1181,68 @@ export function addCommands(
       await dialog.launch();
     }
   });
+  commands.addCommand(CommandIDs.copyObject, {
+    label: trans.__('Copy Object'),
+    isEnabled: () => {
+      const current = tracker.currentWidget;
+      return current ? current.context.model.sharedModel.editable : false;
+    },
+    execute: () => {
+      const current = tracker.currentWidget;
+      if (!current) {
+        return;
+      }
+
+      const objectId = getSelectedObjectId(current);
+      const sharedModel = current.context.model.sharedModel;
+      const objectData = sharedModel.getObjectByName(objectId);
+
+      if (!objectData) {
+        console.warn('Could not retrieve object data.');
+        return;
+      }
+
+      current.context.model.setCopiedObject([objectData]);
```

Comment:
Ah I missed you were using `IJCadModel`, that looks wrong since you're passing a single object.

`setCopiedObject` and `getCopiedObject` should take and return `IJCadObject` instead

Reply:
I'm on it